### ISSUE TITLE
Infinite loop: max number of client connection reached, flushing the accept queue

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#422]
When doing a lot of concurrent requests (more than the `max_connections` configuration option), the proxy enters in an infinite loop and never flush the accept queue.

To reproduce, see #421, I used the same bash script / configuration.

When the proxy is in the infinite loop and the script is stopped (meaning that no more curl process are started and no other requests are made), curl process are stuck at `501` (value given by `ps aux | grep curl`):

```
2018-03-29T10:11:23Z 199151801385571 5846 WRK-00 ERROR  max number of client connection reached, flushing the accept queue
2018-03-29T10:11:23Z 199151801396755 5846 WRK-00 ERROR  max number of client connection reached, flushing the accept queue
2018-03-29T10:11:23Z 199151801407730 5846 WRK-00 ERROR  max number of client connection reached, flushing the accept queue
2018-03-29T10:11:23Z 199151801419485 5846 WRK-00 ERROR  max number of client connection reached, flushing the accept queue
2018-03-29T10:11:23Z 199151801430486 5846 WRK-00 ERROR  max number of client connection reached, flushing the accept queue
```

**Comment by Geal** [comment]
this should be fixed by d542637 6c85167 and f04dc63

**Comment by BlackYoup** [comment]
It seems to be fixed now
